Sources/UtiliKit/ActiveLabel/ActiveLabel.swift

```diff
@@ -0,0 +1,324 @@
+        self.loadingAnimationDelay = configuration.loadingAnimationDelay
+        
+        configurationChanged()
+    }
+    
+    private func commonInit() {
+        textDidUpdate()
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        loadingViews.forEach { [weak self] (view) in
+            guard let self = self else { return }
+            if let maskLayer = view.layer.mask {
+                maskLayer.bounds = CGRect(x: 0, y: 0, width: self.bounds.width + (maskOffset * 2), height: self.bounds.height)
+                maskLayer.frame = CGRect(x: -maskOffset, y: 0, width: self.bounds.width + (maskOffset * 2), height: self.bounds.height)
+            }
+        }
+    }
+    
+    override func prepareForInterfaceBuilder() {
+        super.prepareForInterfaceBuilder()
+        
+        isDisplayingInStoryboard = true
+        state = showLoadingViewsInStoryboard ? .loading : .text("")
+    }
+    
+    /**
+     You must call this function to reset the loading views after you have changed a configuration value.
+     */
+    func configurationChanged() {
+        loadingViews.forEach { $0.removeFromSuperview() }
+        loadingViews.removeAll()
+        
+        if state == .loading {
+            showLoadingViews()
+        }
+    }
+}
+
+private extension ActiveLabel {
+    func textDidUpdate() {
+        state = text == nil ? .loading : .text(text ?? "")
+    }
+    
+    // MARK: - Loading View Functions
+    func showLoadingViews() {
+        if loadingViews.isEmpty {
+            configureLoadingViews()
+        }
+        loadingViews.forEach({ $0.isHidden = false })
+    }
+    
+    func hideLoadingViews() {
+        loadingViews.forEach({ $0.isHidden = true })
+    }
+    
+    func configureLoadingViews() {
+        for index in 0..<Int(estimatedNumberOfLines) {
+            let view = UIView()
+            view.translatesAutoresizingMaskIntoConstraints = false
+            view.backgroundColor = loadingViewColor
+            
+            let maskLayer = createLayerMask()
+            maskLayer.add(maskAnimation(), forKey: "maskAnimation")
+            view.layer.mask = maskLayer
+            loadingViews.append(view)
+            
+            addSubview(view)
+            setupConstraints(for: view, at: index)
+        }
+        
+        loadingViews.forEach({ $0.isHidden = isHidden })
+    }
+    
+    func setupConstraints(for view: UIView, at index: Int) {
+        NSLayoutConstraint.activate([view.leadingAnchor.constraint(equalTo: leadingAnchor),
+                                     view.heightAnchor.constraint(greaterThanOrEqualToConstant: loadingLineHeight)])
+        
+        if index == 0 {
+            // Constrain the first view to the top of the label.
+            view.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        } else {
+            // If not the first view constrain the top to the above views bottom with the loadingLineVerticalSpacing.
```

Comment:
Very minor typo (missing comma and apostrophe).
```swift
// If not the first view, constrain the top to the above view's bottom with the loadingLineVerticalSpacing.
```

Reply:
Fixed.